scripts/core/editor3/helpers/characters-limit.ts

```diff
@@ -0,0 +1,178 @@
+import {
+    EditorState,
+    Modifier,
+    SelectionState,
+    DraftEditorCommand,
+    RichUtils,
+} from 'draft-js';
+import {getEditorFieldCharactersCount} from 'apps/authoring/authoring/components/CharacterCount';
+
+export const LIMIT_CHARACTERS_OVERFLOW_STYLE = 'LIMIT_CHARACTERS_OVERFLOW';
+
+export function preventInputWhenLimitIsPassed(
+    editorState: EditorState,
+    newChars: string,
+    limit: number,
+) {
+    const length =
+        getEditorFieldCharactersCount(editorState.getCurrentContent().getPlainText(), false) +
+        newChars.length -
+        getLengthOfSelectedText(editorState);
+    const overflow = length - limit;
+
+    return overflow > 0;
+}
+
+function getLengthOfSelectedText(editorState: EditorState) {
+    const currentSelection = editorState.getSelection();
+    const isCollapsed = currentSelection.isCollapsed();
+
+    if (isCollapsed) {
+        return 0;
+    }
+
+    let length = 0;
+    const currentContent = editorState.getCurrentContent();
+    const startKey = currentSelection.getStartKey();
+    const endKey = currentSelection.getEndKey();
+    const startBlock = currentContent.getBlockForKey(startKey);
+    const blocksAreTheSame = startKey === endKey;
+    const startBlockTextLength = startBlock.getLength();
+    const startSelectedTextLength =
+        startBlockTextLength - currentSelection.getStartOffset();
+    const endSelectedTextLength = currentSelection.getEndOffset();
+    const keyAfterEnd = currentContent.getKeyAfter(endKey);
+
+    if (blocksAreTheSame) {
+        length +=
+            currentSelection.getEndOffset() - currentSelection.getStartOffset();
+    } else {
+        let currentKey = startKey;
+
+        while (currentKey && currentKey !== keyAfterEnd) {
```

Comment:
It'd be good to use a for loop instead to avoid infinite loop & browser freeze.

Reply:
Agree, I'm gonna commit a simpler way I found to count selection characters